### ISSUE TITLE
Support full Atom (and RSS) spec

### ISSUE THREAD

**Issue by dotandimet** [dotandimet/Mojo-Feed#14]
@mdom said:

> why not just add type to Mojo::Feed? I would even go so far and add all attributes mentioned in the spec. That shouldn't be too much work and have a negligent runtime cost. And even if we worry about that, we could just use a specialised import list to generate a minimal set of accessors or an extended list. I would love to use atoms expired to save updating feeds if they run their course. :)

I say, +1.

**Comment by mdom** [comment]
Okay, i take a swing at this, but it could take me some days. Weekends are family time... :)

**Comment by mdom** [comment]
Okay, it is harder than i expected to create a unified api for all the different feed specifications. It's probably easier to extend `Mojo::Feed`, write some extensions and see what would be an good fit for `Mojo::Feed`. I'm not sure if this issue should be kept open; if you don't planning on working on it, maybe we just close it and i open a new issue if I have attributes to add? What do you think?

**Comment by dotandimet** [comment]
Yeah, I think that the unified/generic feed API Mojo::Feed currently offers can't shift from being a broadest common denominator to being a superset of all Atom/RSS features. The two formats are sufficiently different that they probably require separate parser classes.

What is Atom's expired attribute you mentioned? Is that part of the standard or some extension?

**Comment by mdom** [comment]
Uhm, sorry, i thought i saw some rfc mentioning this extension, but i can't find it in the moment.

**Comment by dotandimet** [comment]
Huh, I found `expired` in the [JSON Feed](https://jsonfeed.org/version/1) spec. I wonder what sources support that.
